app/src/main/java/com/rockthevote/grommet/data/db/dao/RegistrationDao.kt

```diff
@@ -9,12 +9,12 @@ import com.rockthevote.grommet.data.db.model.Registration
 @Dao
 interface RegistrationDao {
     @Query("SELECT * FROM registration")
-    fun getAll(): List<Registration>
+    suspend fun getAll(): List<Registration>
 
     @Insert
-    fun insert(vararg registration: Registration)
+    suspend fun insert(vararg registrations: Registration)
```

Comment:
guessing that you can't use `async` and `await` if the functions aren't declared as `suspend` ?

Seemed odd to me at first  because the actual logic of how/when to suspend is based on the coroutine calling them, but I guess this would function as a hard check in case the developer didn't want them to be able to be suspsended?

Reply:
You can still use async/await with or without suspend. Marking it as `suspend` just helps ensure that the call isn't run on the Main thread: https://medium.com/androiddevelopers/room-coroutines-422b786dc4c5 .

My understanding is that even if the call happens to occur in a function using the Main dispatcher, the execution of the transaction is shifted off the Main thread.